src/main/java/co/rsk/bitcoinj/script/Script.java

```diff
@@ -489,8 +490,51 @@ public Script getScriptSigWithSignature(Script scriptSig, byte[] sigBytes, int i
      * Returns the index where a signature by the key should be inserted. Only applicable to
      * a P2SH scriptSig.
      */
-    public int getSigInsertionIndex(Sha256Hash hash, BtcECKey signingKey) {
-        return this.redeemScriptParser.getSigInsertionIndex(hash, signingKey);
+    public int getSigInsertionIndex(Sha256Hash hashForSignature, BtcECKey signingKey) {
+        // Iterate over existing signatures, skipping the initial OP_0, the final redeem script
+        // and any placeholder OP_0 sigs.
+
+
+        // In order to affect consensus, we keep returning zero as the default value for the insertion index
+        // when the script is not a scriptSig.
```

Comment:
nit: i would say `To keep backwards compatibility` instead of `In order to affect consensus`

Reply:
Thanks, it was wrong anyways. There was a missing "not" between "to ... affect"

Reply:
jaja right :)

Reply:
Updated